spec/lib/jobs/generate-tag-spec.js

```diff
@@ -0,0 +1,243 @@
+// Copyright 2016, EMC, Inc.
+/* jshint node:true */
+
+'use strict';
+
+describe("Job.Catalog.GenerateTag", function () {
+    var waterline = {};
+    var taskProtocol = {};
+    var GenerateTag;
+    var uuid;
+
+    var catalog1 = {
+        id: 'abf9d5a4ebf61b6715dbaef6',
+        source: 'dmi',
+        data: {
+            dmi: {
+                system: {
+                    manufacturer: 'Renasar',
+                    'product_name': '1A'
+                }
+            }
+        }
+    };
+
+    var catalog2 = {
+        id: 'abf9d5a4ebf61b6715dbaef7',
+        source: 'bmc',
+        data: {
+            'MAC Address' : '00:00:00:00:00:00',
+            'Subnet Mask' : '255.255.255.0',
+            'IP Address' : '10.1.1.1'
+        }
+    };
+
+    before(function () {
+        // create a child injector with on-core and the base pieces we need to test this
+        helper.setupInjector([
+            helper.require('/spec/mocks/logger.js'),
+            helper.require('/lib/jobs/base-job.js'),
+            helper.require('/lib/jobs/generate-tag.js'),
+            helper.require('/lib/utils/job-utils/catalog-searcher.js'),
+            helper.di.simpleWrapper(waterline, 'Services.Waterline'),
+            helper.di.simpleWrapper(taskProtocol, 'Protocol.Task'),
+        ]);
+
+        GenerateTag = helper.injector.get('Job.Catalog.GenerateTag');
+        uuid = helper.injector.get('uuid');
+        waterline.tags = {
+            find: sinon.stub()
+        };
+        waterline.catalogs = {
+            findMostRecent: sinon.stub()
+        };
+        waterline.nodes = {
+            addTags: sinon.stub().resolves()
+        };
+    });
+
+    beforeEach(function () {
+        waterline.tags.find.reset();
+        waterline.catalogs.findMostRecent.reset();
+        waterline.nodes.addTags.reset();
+        taskProtocol.subscribeActiveTaskExists = sinon.stub().resolves({
+            dispose: sinon.stub()
+        });
+    });
+
+    it('assigns a matching tag', function() {
```

Comment:
Humor me, but if you have the time would you add a simple test that does a regex match rather than an equals match? Just so anyone looking for tag definition examples can see one of that usage.